packages/vue-app/template/App.js

```diff
@@ -98,7 +98,18 @@ export default {
   },
   created () {
     // Add this.$nuxt in child instances
-    Vue.prototype.<%= globals.nuxt %> = this
+    if (process.server) {
+      this.$root.$options.<%= globals.nuxt %> = this
+      Object.defineProperty(Vue.prototype, '<%= globals.nuxt %>', {
```

Comment:
I think other than options assignment, this part can be moved in a global context for defining getter